src/ng/parse.js

```diff
@@ -1260,7 +1261,7 @@ ASTCompiler.prototype = {
   },
 
   nonComputedMember: function(left, right) {
-    var SAFE_IDENTIFIER = /[$_a-zA-Z][$_a-zA-Z0-9]*/;
+    var SAFE_IDENTIFIER = /^[$_a-zA-Z][$_a-zA-Z0-9]$/;
```

Comment:
Don't we need to keep the `*`? Otherwise almost everything will use the `['...']` format...